src/components/FormStep.js

```diff
@@ -70,13 +96,38 @@ const FormStep = ({ form, submission, onStepSubmitted, onLogout }) => {
     [step.url]
   );
 
+  useDebounce(
+    async () => {
+      const data = state.data;
+      if (!data) return;
+
+      dispatch({type: 'BLOCK_SUBMIT'});
+      const submissionData = await doLogicCheck(step.url, data);
+      onReloadSubmission(submissionData.submission);
+
+      // TODO: check custom attributes for submission button control
+      const formInstance = formRef.current.instance.instance;
+
+      // we can't just dispatch this, because Formio keeps references to DOM nodes
+      // which expire when the component re-renders, and that gives React
+      // unstable_flushDiscreteUpdates warnings. However, we can update the form
+      // definition by using the ref to the underlying Formio instance.
+      formInstance.setForm(submissionData.step.formStep.configuration);
+      dispatch({type: 'ENABLE_SUBMIT'});
```

Comment:
I'm not sure about this - what happens if the action is to disable the step submission?

Reply:
Mhmm  good point :thinking: